internal/util.go

```diff
@@ -12,3 +16,10 @@ func CombineErrors(ret error, also error, desc string) error {
 	}
 	return ret
 }
+
+// Report errors in defers
+func LogDeferred(logger log.Logger, f func() error) {
```

Comment:
```suggestion
//LogDeferred logs an error in a deferred function call if the returned error is non-nil.
func LogDeferred(logger log.Logger, f func() error) {
```

Reply:
Also, why is this defined in `internal`?

Reply:
I don't have an specific reason for that, I thought was a good place for this, but if you have a suggestion for a better place I would be happy to change it.

Reply:
Comment changed.

Reply:
Well if I'm not mistaken, this is under `internal` which means consumers of the SDK are not able to utilize this API. Maybe `types/utils.go` for now?

Reply:
No, no, you're right, I just didn't think the intention was to give SDK consumers access to this feature. I've already moved it to types/util.go.